src/fw-logs/fw-log-data.cpp

```diff
@@ -1,68 +1,63 @@
 // License: Apache 2.0. See LICENSE file in root directory.
-// Copyright(c) 2019 Intel Corporation. All Rights Reserved.
+// Copyright(c) 2024 Intel Corporation. All Rights Reserved.
+
 #include "fw-log-data.h"
-#include <sstream>
-#include <iomanip>
-#include <locale>
-#include <string>
 
-using namespace std;
+#include <rsutils/string/from.h>
 
-# define SET_WIDTH_AND_FILL(num, element) \
-setfill(' ') << setw(num) << left << element \
 
 namespace librealsense
 {
     namespace fw_logs
     {
-        rs2_log_severity fw_log_data::get_severity() const
-        {
-            return fw_logs_severity_to_log_severity(_severity);
-        }
-
-        const std::string& fw_log_data::get_message() const
-        {
-            return _message;
-        }
-
-        const std::string& fw_log_data::get_file_name() const
-        {
-            return _file_name;
-        }
-
-        const std::string& fw_log_data::get_thread_name() const
-        {
-            return _thread_name;
-        }
-
-        uint32_t fw_log_data::get_line() const
-        {
-            return _line;
-        }
-
-        uint32_t fw_log_data::get_timestamp() const
+        rs2_log_severity fw_logs_binary_data::get_severity() const
         {
-            return (uint32_t)_timestamp;
-        }
+            const fw_log_binary * log_binary = reinterpret_cast< const fw_log_binary * >( logs_buffer.data() );
+            if( log_binary->magic_number == 0xA5)
+                return fw_logs_severity_to_rs2_log_severity( log_binary->severity );
+            if( log_binary->magic_number == 0xA0 )
+                return legacy_fw_logs_severity_to_rs2_log_severity( log_binary->severity );
 
-        uint32_t fw_log_data::get_sequence_id() const
-        {
-            return _sequence;
+            throw librealsense::invalid_value_exception( rsutils::string::from()
+                                                         << "Received unfamiliar FW log 'magic number' "
+                                                         << log_binary->magic_number );
         }
 
-        rs2_log_severity fw_logs_binary_data::get_severity() const
+        uint32_t fw_logs_binary_data::get_timestamp() const
         {
-            const fw_log_binary* log_binary = reinterpret_cast<const fw_log_binary*>(logs_buffer.data());
-            return fw_logs_severity_to_log_severity(static_cast<int32_t>(log_binary->dword1.bits.severity));
+            const legacy_fw_log_binary * log_binary = reinterpret_cast< const legacy_fw_log_binary * >( logs_buffer.data() );
+            return log_binary->timestamp;
```

Comment:
Can't we use the previous `_timestamp` here?

Reply:
Missed this. I need to update this to support the new format also (like in `get_severity`).
Legacy and extended format timestamp is different - other location, type, and the extended has 2 timestamps, we might need to output both.

Reply:
Done